plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2932,17 +2936,31 @@ public MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession s
                 .orElseThrow(() -> new IllegalStateException("Storage table missing in definition of materialized view " + materializedViewName));
 
         Table icebergTable = catalog.loadTable(session, storageTableName);
-        String dependsOnTables = Optional.ofNullable(icebergTable.currentSnapshot())
+        Snapshot currentSnapshot = icebergTable.currentSnapshot();
+        String dependsOnTables = Optional.ofNullable(currentSnapshot)
                 .map(snapshot -> snapshot.summary().getOrDefault(DEPENDS_ON_TABLES, ""))
                 .orElse("");
+        String dependsOnTableFunctions = Optional.ofNullable(icebergTable.currentSnapshot())
```

Comment:
why not use currentSnapshot variable you extracted?

Reply:
accidental mistake